libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -0,0 +1,236 @@
+#include <AP_HAL/AP_HAL.h>
+
+#if AP_DDS_ENABLED
+
+#include <AP_RTC/AP_RTC.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+#include <GCS_MAVLink/GCS.h>
+
+#include "AP_DDS_Client.h"
+#include "AP_DDS_Gather_Data.h"
+#include "generated/Time.h"
+
+AP_HAL::UARTDriver *dds_port;
+
+
+const AP_Param::GroupInfo AP_DDS_Client::var_info[]= {
+    //! @todo Params go here
+
+    AP_GROUPEND
+};
+
+#include "AP_DDS_Topic_Table.h"
+
+/*
+  class constructor
+ */
+AP_DDS_Client::AP_DDS_Client(void)
+{
+    if (!hal.scheduler->thread_create(FUNCTOR_BIND_MEMBER(&AP_DDS_Client::main_loop, void),
+                                      "DDS",
+                                      8192, AP_HAL::Scheduler::PRIORITY_IO, 1)) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"DDS Client: thread create failed");
+    }
+}
+
+/*
+  main loop for DDS thread
+ */
+void AP_DDS_Client::main_loop(void)
+{
+    if (!init() || !create()) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"DDS Client: Creation Requests failed");
+        return;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Initialization passed");
+    while (true) {
+        hal.scheduler->delay(1);
+        update();
+    }
+}
+
+
+bool AP_DDS_Client::init()
+{
+    AP_SerialManager *serial_manager = AP_SerialManager::get_singleton();
+    dds_port = serial_manager->find_serial(AP_SerialManager::SerialProtocol_DDS_XRCE, 0);
+    if (dds_port == nullptr) {
+        return false;
+    }
+
+    // ensure we own the UART
+    dds_port->begin(0);
+
+    constexpr uint8_t fd = 0;
+    constexpr uint8_t relativeSerialAgentAddr = 0;
+    constexpr uint8_t relativeSerialClientAddr = 1;
+    if (!uxr_init_serial_transport(&serial_transport,fd,relativeSerialAgentAddr,relativeSerialClientAddr)) {
+        return false;
+    }
+
+    constexpr uint32_t uniqueClientKey = 0xAAAABBBB;
+    //TODO does this need to be inside the loop to handle reconnect?
+    uxr_init_session(&session, &serial_transport.comm, uniqueClientKey);
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Initialization wait");
+    while (!uxr_create_session(&session)) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Initialization waiting...");
+        hal.scheduler->delay(1000);
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Session Created");
+
+    reliable_in = uxr_create_input_reliable_stream(&session,input_reliable_stream,BUFFER_SIZE_SERIAL,STREAM_HISTORY);
+    reliable_out = uxr_create_output_reliable_stream(&session,output_reliable_stream,BUFFER_SIZE_SERIAL,STREAM_HISTORY);
+
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Init Complete");
+
+    return true;
+}
+
+bool AP_DDS_Client::create()
+{
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: create() enter");
```

Comment:
i think we can remove some of these debug messages now

Reply:
Done 597edf87f44fafdd208fd336e6cf9104d044ad44

Only the error ones, and final success message exist.